src/components/common/PeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -282,24 +268,85 @@ export default class PeopleAndRoles extends Mixins(CommonMixin, DateMixin) {
-    }
-    if (this.isChangeFiling || this.isConversionFiling) {
-      if (this.isTypeSoleProp) return this.hasMinimumProprietor
-      if (this.isTypePartnership) return this.hasMinimumPartners
+    switch (true) {
+      case (this.isAlterationFiling): {
+        // alterations don't use this component
+        return false
+      }
+      case (this.isChangeRegFiling): {
+        if (this.isTypeSoleProp) return this.hasMinimumProprietor
+        if (this.isTypePartnership) return this.hasMinimumPartners
+        return false
+      }
+      case (this.isConversionFiling): {
+        if (this.isTypeSoleProp) return this.hasMinimumProprietor
+        if (this.isTypePartnership) return this.hasMinimumPartners
+        return false
+      }
+      case (this.isCorrectionFiling): {
+        if (this.isTypeBcomp) return (this.cpValid && this.incorpValid && this.dirValid)
+        return false
+      }
     }
-    return false // should never get here
+    return false // should never happen
   }
 
-  /** True if there are no orgs/persons with missing roles. */
+  /**
+   * True if all parties have the required addresses.
+   * See also OrgPerson.vue::mailingAddressSchema and deliveryAddressSchema.
+   */
+  get haveRequiredAddresses (): boolean {
+    return this.getPeopleAndRoles.every(party => {
+      // NB: some parties have multiple roles, so order matters below
+      //     (most restrictive to least restrictive)
+
+      // proprietor/partner must have a mailing address and a delivery address
+      // delivery address must be in BC, Canada
+      if (this.hasRoleProprietor(party) || this.hasRolePartner(party)) {
+        return (
+          !isEmpty(party.mailingAddress) &&
+          !isEmpty(party.deliveryAddress) &&
+          (party.deliveryAddress.addressRegion === REGION_BC) &&
+          (party.deliveryAddress.addressCountry === COUNTRY_CA)
+        )
+      }
+
+      // director must have a mailing address and a delivery address
+      if (this.hasRoleDirector(party)) {
+        return (
+          !isEmpty(party.mailingAddress) &&
+          !isEmpty(party.deliveryAddress)
+        )
+      }
+
+      // completing party must have just a mailing address
+      if (this.hasRoleCompletingParty(party)) {
+        return !isEmpty(party.mailingAddress)
+      }
+
+      // incorporators must have just a mailing address
+      if (this.hasRoleIncorporator(party)) {
+        return !isEmpty(party.mailingAddress)
+      }
+    })
+  }
+
+  /** True if all orgs/persons have a role. */
   get noMissingRoles (): boolean {
     return this.getPeopleAndRoles.every(p => p.roles.length > 0)
   }
 
+  /** True if OrgPersons list is valid. */
+  get validOrgPersons (): boolean {
+    // not in editing mode and all requirements are met
+    return (
+      !this.isAddingEditingOrgPerson &&
+      this.haveRequiredParties &&
+      this.haveRequiredAddresses &&
+      this.noMissingRoles
+    )
+  }
```

Comment:
This is the validity getter for the entire People And Roles component (used for corrections, changes and conversions). It uses other getters (declared above) to check the list and ensure it's valid.